components/sensor/tsl2591.rst

```diff
@@ -0,0 +1,199 @@
+TSL2591 Ambient Light Sensor
+============================
+
+.. seo::
+    :description: Instructions for setting up TSL2591 ambient light sensors in ESPHome.
+    :image: tsl2591.jpg
+    :keywords: TSL2591
+
+The ``tsl2591`` sensor platform allows you to use the AMS TSL2591 ambient light sensor with ESPHome.
+Communication with the device is over :ref:`I²C <i2c>`, which must be present in your configuration.
+
+The TSL2591 device is available on breakout boards from a few vendors
+(for example, `Adafruit`_, `CQRobot`_, `Waveshare`_).
+
+.. _Adafruit: http://www.adafruit.com/products/1980
+.. _CQRobot: https://www.cqrobot.com/index.php?route=product/product&product_id=1112
+.. _Waveshare: https://www.waveshare.net/shop/TSL25911-Light-Sensor.htm
+
+.. figure:: images/tsl2591.jpg
+    :align: center
+    :width: 50.0%
+
+    TSL2591 Ambient Light Sensor on a breakout board.
+
+.. figure:: images/tsl2591-ui.jpg
+    :align: center
+    :width: 100.0%
+
+    TSL2591 sensors in Home Assistant UI.
+
+The sensor claims a dynamic range of 600 million to 1 with an effective maximum of 88000 lux.
+It achieves that large range by having a configurable ``gain`` value.
+Use a higher gain value when measuring less intense light sources.
+On the other hand, if you get ADC readings of 65,535 for either physical sensor,
+you may be saturating that sensor and need to reduce the gain.
+This Wikipedia `article <https://en.wikipedia.org/wiki/Lux>`__ has a table of some lux values for comparison.
+
+The implementation offers four logical sensors, each of which has a default unit of measurement of ``lux``.
+Only one of them, **calculated_lux**, is a true ``lux`` value, and even that is a bit subjective.
+(For information about the difference between radiometric and photometric measurements, see the references.)
+The device's two physical sensors measure "visible and infrared" (channel 0) and "infrared" (channel 1), respectively.
+However, the logical sensor readings here also provide the simple "visible" value separately.
+The visible value is obtained by subtracting the reading of the physical infrared sensor
+(channel 1) from the value of the physical sensor that combines visible and infrared (channel 0).
+Since those are two different infrared sensor readings, there is the possibility of a small inaccuracy.
+
+- **full_spectrum**: Raw 16 bit reading from the on-board ADC for the physical sensor for visible and infrared light (channel 0).
+  This is not really ``lux``.
+- **infrared**: Raw 16 bit reading from the on-board ADC for the physical sensor for infrared light (channel 1).
+  This is not really ``lux``
+- **visible**: For convenience, a value calculated by taking the difference between the two physical sensors (channel 0 minus channel 1).
+  This is not really ``lux``.
```

Comment:
If this is not really the lux unit, maybe we should change the unit of measurement of the sensor? I think it's possible to provide raw values without any unit of measurement (but I haven't tested that).

Reply:
Yeah, I have been kind of undecided about that, but I think you are right that it's less confusing if they are unitless. Both ESPHome and Home Assistant seem happy with UNIT_EMPTY. I'm going to keep the same icon, though, since they do indicate relative brightness.

I'll make this change in the next update to the PR.